src/peer_map.rs

```diff
@@ -6,24 +6,13 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-use crate::{id::PublicId, xor_name::XorName, ConnectionInfo};
+use crate::ConnectionInfo;
 use fxhash::{FxHashMap as HashMap, FxHashSet as HashSet};
-use std::{collections::hash_map::Entry, net::SocketAddr};
+use std::net::SocketAddr;
 
-/// This structure holds the bi-directional association between peers public id and their network
-/// connection info. This association can be create in two ways:
-/// 1. When both pieces of information (public id and connection info) are obtained at the same
-///    time, call `insert`. This happens when a third party (other members of the section) sends
-///    them to us.
-/// 2. Otherwise its a two step process: first, when the connection to the peer is established at
-///    the network layer, call `connect`. Then when their public id is received, call `identify`.
-///    This happens when the peer connects to us and then sends us a message which contains their
-///    public id.
 #[derive(Default)]
 pub struct PeerMap {
-    forward: HashMap<XorName, ConnectionInfo>,
-    reverse: HashMap<SocketAddr, HashSet<PublicId>>,
-    pending: HashMap<SocketAddr, PendingConnection>,
+    connections: HashMap<SocketAddr, ConnectionInfo>,
```

Comment:
Now the socket address is stored twice - in the key and in the value. Could we change this to `HashMap<SocketAddr, Certificate>` ?

Reply:
This was indeed discussed during planning. The argument for `ConnectionInfo` was that you can return a reference and just use that directly. I think both ways has merit. What do you think @jeanphilippeD ?

Reply:
We had a chat before, this is simpler and the gain is very small, that avoid having to rebuild the ConnectionInfo everytime we get it.

Reply:
OK, fair enough. I think at some point in the future we should change the way we store the certificate from `Vec<u8>` to `Rc<[u8]>` (or `Arc<[u8]>`) to make them cheap to clone. Then it wouldn't matter to return them by value instead of reference so we would be able to avoid this duplication.